x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformConfig.java

```diff
@@ -449,12 +449,17 @@ public static TransformConfig fromXContent(final XContentParser parser, @Nullabl
     public static TransformConfig rewriteForUpdate(final TransformConfig transformConfig) {
 
         // quick checks for deprecated features, if none found just return the original
-        if (transformConfig.getPivotConfig() == null || transformConfig.getPivotConfig().getMaxPageSearchSize() == null) {
+        if (transformConfig.getVersion() != null
+            && transformConfig.getVersion().onOrAfter(Version.V_8_0_0) // todo: V_7_11_0
+            && (transformConfig.getPivotConfig() == null || transformConfig.getPivotConfig().getMaxPageSearchSize() == null)) {
```

Comment:
Is this deprecation related to handling dates?

Reply:
ok, the code comment is outdated, will change it. What this does is basically rewriting the config to the latest format. The problem is as follows, a transform `<7.11` will write millis, `7.11`> will write strings (note the version is the version of the transform, not the es version). If update is called, it always updates the version field, a former `7.6` transform becomes a `7.11` transform. But that would mean that this transform starts writing strings. That's why we inject the `dates_as_epoch_millis` into the config, so it still becomes a `7.11` transform config, but with `dates_as_epoch_millis:true`.

Reply:
ok, the code comment is outdated, will change it